zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -196,14 +199,50 @@ public ZooKeeperServer(FileTxnSnapLog txnLogFactory, int tickTime,
         setMinSessionTimeout(minSessionTimeout);
         setMaxSessionTimeout(maxSessionTimeout);
         listener = new ZooKeeperServerListenerImpl(this);
+
         readResponseCache = new ResponseCache();
+
+        connThrottle = new BlueThrottle();
```

Comment:
Agree. Those setter invocations in the constructor look ugly. Moved them out. But we still need the setters for the JMX interface. And since we are keeping setters, I'd prefer to use them in the unit tests instead of using a constructor with a long list of parameters, i.e, I prefer to explicitly set the refill time to 100 instead of setting the third parameter of the constructor to 100.